src/main/java/org/folio/circulation/domain/CreateRequestService.java

```diff
@@ -19,65 +18,66 @@
 
     this.requestRepository = requestRepository;
     this.updateItem = updateItem;
     this.updateLoanActionHistory = updateLoanActionHistory;
     this.updateLoan = updateLoan;
     this.requestPolicyRepository = requestPolicyRepository;
+    this.loanRepository = loanRepository;
   }
 
   public CompletableFuture<Result<RequestAndRelatedRecords>> createRequest(
-    RequestAndRelatedRecords requestAndRelatedRecords) {
-
-    return completedFuture(refuseWhenItemDoesNotExist(requestAndRelatedRecords)
-      .next(CreateRequestService::refuseWhenInvalidUserAndPatronGroup)
-      .next(CreateRequestService::refuseWhenItemIsNotValid)
-      .next(CreateRequestService::refuseWhenUserHasAlreadyRequestedItem))
-      .thenComposeAsync(r -> r.after(requestPolicyRepository::lookupRequestPolicy))
-      .thenApply(r -> r.next(CreateRequestService::refuseWhenRequestCannotBeFulfilled))
-      .thenApply(r -> r.map(CreateRequestService::setRequestQueuePosition))
-      .thenComposeAsync(r -> r.after(updateItem::onRequestCreation))
-      .thenComposeAsync(r -> r.after(updateLoanActionHistory::onRequestCreation))
-      .thenComposeAsync(r -> r.after(updateLoan::onRequestCreation))
-      .thenComposeAsync(r -> r.after(requestRepository::create));
+      RequestAndRelatedRecords requestAndRelatedRecords) {
+
+    return findOpenLoanWithRequest(requestAndRelatedRecords, loanRepository)
+      .thenApply(loanResult ->
+         refuseWhenUserHasAlreadyBeenLoanedItem(requestAndRelatedRecords, loanResult)
+           .next(CreateRequestService::refuseWhenItemDoesNotExist)
+           .next(CreateRequestService::refuseWhenInvalidUserAndPatronGroup)
+           .next(CreateRequestService::refuseWhenItemIsNotValid)
+           .next(CreateRequestService::refuseWhenUserHasAlreadyRequestedItem)
+       ).thenComposeAsync(r -> r.after(requestPolicyRepository::lookupRequestPolicy))
+        .thenApply(r -> r.next(CreateRequestService::refuseWhenRequestCannotBeFulfilled))
+        .thenApply(r -> r.map(CreateRequestService::setRequestQueuePosition))
+        .thenComposeAsync(r -> r.after(updateItem::onRequestCreation))
+        .thenComposeAsync(r -> r.after(updateLoanActionHistory::onRequestCreation))
+        .thenComposeAsync(r -> r.after(updateLoan::onRequestCreation))
+        .thenComposeAsync(r -> r.after(requestRepository::create));
   }
 
-  private static RequestAndRelatedRecords setRequestQueuePosition(
-    RequestAndRelatedRecords requestAndRelatedRecords) {
+  private static RequestAndRelatedRecords setRequestQueuePosition(RequestAndRelatedRecords requestAndRelatedRecords) {
 
-    //TODO: Extract to method to add to queue
+    // TODO: Extract to method to add to queue
     requestAndRelatedRecords.withRequest(requestAndRelatedRecords.getRequest()
-      .changePosition(requestAndRelatedRecords.getRequestQueue().nextAvailablePosition()));
+        .changePosition(requestAndRelatedRecords.getRequestQueue().nextAvailablePosition()));
```

Comment:
This appears to be 4 space indentation, is there a chance the IDE is automatically indenting by 4 rather than 2 spaces?

Reply:
I will check to make sure it is not